routes/devices.js

```diff
@@ -120,15 +121,139 @@ devices.post('/:id', auth.ensureAuthenticated, function (req, res) {
 devices.get('/:id/json', auth.ensureAuthenticated, function (req, res) {
   Device.findById(req.params['id'], function (err, device) {
     if (err) {
-      return res.status(404).render('error', {
-        error: {
-          status: err
-        }
-      });
+      log.error(err);
+      return res.status(500).send(err.message);
     }
     res.status(200).json(device);
   });
 });
 
+devices.get('/json/serialNos', auth.ensureAuthenticated, function (req, res) {
+  Device.find({}, {serialNo: true}, function (err, docs) {
+    if (err) {
+      log.error(err);
+      return res.status(500).send(err.message);
+    }
+    return res.status(200).json(docs);
+  });
+});
+
+
+/**
+ * update status and installToDevice for device
+ * set prepare: status 0 --> 1 and installToDevice nul --> id
+ * set spare: status (1|1.5|2|3) --> 0 and installToDevice id--> null
+ */
+devices.put('/:id/installToDevice/:oldId/:newId/status/:oldStatus/:newStatus', auth.ensureAuthenticated, checkStatusTransition, function (req, res) {
```

Comment:
the client should not be able to change the target and status at the same time. 
this needs to be seperated into two routes
put /:id/installtodevice and include the target id in the body. if current the device was to be installed or already installed in another device, then should send 4xx back and explain. 

put /:id/status is similar, the status should be in body. The route validate the logic of status change. if the change is not valid, send 4xx and explain.